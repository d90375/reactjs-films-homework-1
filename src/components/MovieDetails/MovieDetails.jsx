import React from 'react';
import MovieDetailsInfo from '../MovieDetailsInfo';
import MovieDetailsDescription from '../MovieDetailsDescription';
import MovieDetailsRating from '../MovieDetailsRating';
import Button from '../Button';

import styles from './MovieDetails.scss';

const data = {
  title: 'The Jungle Book',
  genres: ['Adventure', 'Drama', 'Family', 'Fantasy'],
  duration: '1h 46m',
  description: `There are growing dangers in the wizarding world of 1926 New York.  
                Something mysterious is leaving a path of destruction in the streets, 
                threatening to expose the wizarding`,
  rating: '4.8',
};

const MovieDetailsPage = () => {
  const { description, rating, ...info } = data;

  return (
    <section className={styles.container}>
      <div className={styles.info}>
        <MovieDetailsInfo info={info} />
        <MovieDetailsDescription description={description} />
      </div>
      <div className={styles.info}>
        <MovieDetailsRating rating={rating} />
        <div className={styles.buttonsContainer}>
          <Button purpose="watch">Watch Now</Button>
          <Button purpose="view">View Info</Button>
        </div>
      </div>

    </section>
  );
};

export default MovieDetailsPage;
