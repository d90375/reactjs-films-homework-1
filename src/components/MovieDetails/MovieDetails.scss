.container {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  height: 45vw;
  max-height: 1100px;
  padding: 70px 5% 50px 5%;
  background-color: rgb(0, 0, 0);
  background-image: url('assets/images/junglebook.png');
  background-size: cover;

  @media (max-width: 1024px) {
    height: auto;
    background-size: auto 100%;
    background-position: top center;
  }

  @media (max-width: 767px) {
    padding-top: 140px;
  }
}

.info {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;

  @media (max-width: 1024px) {
    flex-direction: column;
    align-items: center;
  }
}

.buttonsContainer {
  display: flex;
  justify-content: space-around;
  width: 250px;
  padding-top: 20px;
}
