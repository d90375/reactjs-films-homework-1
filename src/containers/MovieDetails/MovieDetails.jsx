import React from 'react';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';

import MovieInfo from '../../components/MovieInfo';
import MovieRating from '../../components/MovieRating';
import MovieAction from '../../components/MovieAction';
import Spinner from '../../components/Spinner';

import styles from './MovieDetails.scss';

const MovieDetails = ({
  details, error, isLoading, fetchDetails,
}) => {
  const { id } = useParams();

  if (!details && !isLoading) {
    fetchDetails(id);
  }

  if (error) {
    return (
      <section className={styles.container}>{`Error! ${error.message}`}</section>
    );
  }

  if (details) {
    const {
      background, rating, overview, ...info
    } = details;

    const style = { backgroundImage: `url(${background})` };
    return (
      <section className={styles.container} style={style}>
        <div className={styles.info}>
          <MovieInfo info={info} />
          <MovieRating rating={rating} />
        </div>
        <MovieAction description={overview} />
      </section>
    );
  }

  return (
    <section className={styles.container}>
      <Spinner />
    </section>
  );
};

MovieDetails.propTypes = {
  fetchDetails: PropTypes.func.isRequired,
  error: PropTypes.shape({
    message: PropTypes.string.isRequired,
  }),
  isLoading: PropTypes.bool.isRequired,
  details: PropTypes.shape({
    background: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    overview: PropTypes.string.isRequired,
  }),
};

MovieDetails.defaultProps = {
  details: null,
  error: null,
};

export default MovieDetails;
