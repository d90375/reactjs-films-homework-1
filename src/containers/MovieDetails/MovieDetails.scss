.container {
  display: flex;
  justify-content: space-between;
  height: 65vh;
  max-height: 1100px;
  padding: 70px 5% 0 5%;
  background-size: cover;

  @media (max-width: 767px) {
    flex-direction: column;
    align-items: center;
    padding: 140px 5% 0;
    height: auto;
    background-size: auto 100%;
    background-position: top center;
  }

  & + section {
    padding-top: 20px;
  }
}

.info {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  flex-grow: 1;
  padding: 15px 15px 50px 15px;
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0%, rgba(0, 0, 0, 0.1) 60%, rgba(0, 0, 0, 0.7) 100%);

  @media (max-width: 767px) {
    width: 100%;
    align-items: center;
    background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0%, rgba(0, 0, 0, 0.7) 100%);
  }
}
