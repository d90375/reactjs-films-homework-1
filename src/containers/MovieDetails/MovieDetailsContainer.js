import { connect } from 'react-redux';
import { getDetails, getDetailsIsLoading, getDetailsError } from '../../modules/details/detailsSelector';
import { fetchDetails } from '../../modules/details/detailsAction';

import MovieDetails from './MovieDetails';

export const mapStateToProps = (state) => ({
  details: getDetails(state),
  error: getDetailsError(state),
  isLoading: getDetailsIsLoading(state),
});

const mapDispatchToProps = {
  fetchDetails,
};


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MovieDetails);
