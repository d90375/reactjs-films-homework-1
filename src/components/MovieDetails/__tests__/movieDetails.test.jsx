import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetails from '../MovieDetails';

const data = {
  description: 'description',
  rating: '5',
};

test('MovieDetails renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieDetails data={data} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
