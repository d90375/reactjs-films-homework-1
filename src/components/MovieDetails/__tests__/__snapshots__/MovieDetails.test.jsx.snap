// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<section
  className="container"
>
  <div
    className="info"
  >
    <MovieInfo
      info={Object {}}
    />
    <MovieRating
      rating={5}
    />
  </div>
  <MovieAction
    description="description"
  />
</section>
`;
